src/transport_alveo.py

```diff
@@ -412,3 +482,220 @@ def get_hgbe_counters(self):
+    AM_RO_AGE_LIMIT_PS1 - Number of times the reorder queue entry reaches its age limit in the last monitoring interval.
+    AM_RMW_CYCLE_PS0 - Number of Read Modify Write cycles in the last monitoring interval.
+    AM_RMW_CYCLE_PS1 - Number of Read Modify Write cycles in the last monitoring interval.
+    
+    '''
+    
+    hbm_stacks = []        
+
+    if stack == None and controller == None and reg_name == '':
+        widgets = ['Processing: ', progressbar.AnimatedMarker()]
+        bar = progressbar.ProgressBar(widgets=widgets).start()
+
+        for hbm_stack_idx in range(2):
+            hbm_base_addr = self.get_hbm_base_addr(hbm_stack_idx)
+            controllers = []
+            self.logger.info("stack " + str(hbm_stack_idx))
+            for region_idx, upper_5_bit_addr in self.hbm_controller_regions.items():
+                registers = dict()
+                self.logger.info(" reading memory controller " + str(region_idx))
+                i = 0
+                for reg, reg_offset in self.hbm_reg_map.items():
+                    registers[reg] = self.memread(hbm_base_addr + (reg_offset + (upper_5_bit_addr << 17)))
+                    bar.update(i)
+                    i = i + 1
+                controllers.insert(region_idx, registers)
+                hbm_stacks.insert(hbm_stack_idx, controllers)
+        return hbm_stacks
+
+    elif stack != None and controller == None and reg_name == '':
+        widgets = ['Processing: ', progressbar.AnimatedMarker()]
+        bar = progressbar.ProgressBar(widgets=widgets).start()
+
+        hbm_base_addr = self.get_hbm_base_addr(stack)
+        controllers = []
+        self.logger.info("stack " + str(stack))
+        for region_idx, upper_5_bit_addr in self.hbm_controller_regions.items():
+            registers = dict()
+            self.logger.info(" reading memory controller " + str(region_idx))
+            i = 0
+            for reg, reg_offset in self.hbm_reg_map.items():
+                registers[reg] = self.memread(hbm_base_addr + (reg_offset + (upper_5_bit_addr << 17)))
+                bar.update(i)
+                i = i + 1
+            controllers.insert(region_idx, registers)
+        return controllers
+
+    elif stack != None and controller != None and reg_name == '':
+        hbm_base_addr = self.get_hbm_base_addr(stack)
+        registers = dict()
+        for reg, reg_offset in self.hbm_reg_map.items():
+                registers[reg] = self.memread(hbm_base_addr + (reg_offset + (self.hbm_controller_regions.get(controller) << 17)))
+        return registers
+
+    elif stack != None and controller != None and reg_name != '':
+        hbm_base_addr = self.get_hbm_base_addr(stack)
+        registers = dict()
+        registers[reg_name] = self.memread(hbm_base_addr + (self.hbm_reg_map.get(reg_name) + (self.hbm_controller_regions.get(controller) << 17)))
+        return registers
+
+    else:
+        self.logger.error('Arguments error')
```

Comment:
You should give more information about what is wrong with the arguments.

Reply:
I have provided more detail to the 'Arguments error'.